core/src/main/java/tc/oc/pgm/shops/ShopModule.java

```diff
@@ -0,0 +1,217 @@
+package tc.oc.pgm.shops;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.SetMultimap;
+import com.google.common.collect.Sets;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import org.bukkit.Material;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Villager;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.util.Vector;
+import org.jdom2.Attribute;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.MapTag;
+import tc.oc.pgm.api.map.factory.MapFactory;
+import tc.oc.pgm.api.map.factory.MapModuleFactory;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.kits.ItemKit;
+import tc.oc.pgm.kits.Kit;
+import tc.oc.pgm.kits.KitParser;
+import tc.oc.pgm.shops.menu.Category;
+import tc.oc.pgm.shops.menu.Icon;
+import tc.oc.pgm.util.material.Materials;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.XMLUtils;
+
+public class ShopModule implements MapModule {
+
+  private static final Collection<MapTag> TAGS =
+      ImmutableList.of(new MapTag("shops", "Shops", false, true));
+
+  private final Set<Shop> shops = Sets.newHashSet();
+  private final Set<ShopKeeper> shopKeepers = Sets.newHashSet();
+
+  @Override
+  public MatchModule createMatchModule(Match match) {
+    return new ShopMatchModule(match, shops, shopKeepers);
+  }
+
+  @Override
+  public Collection<MapTag> getTags() {
+    return TAGS;
+  }
+
+  public static class Factory implements MapModuleFactory<ShopModule> {
+
+    @Override
+    public ShopModule parse(MapFactory factory, Logger logger, Document doc)
+        throws InvalidXMLException {
+      ShopModule module = new ShopModule();
+      KitParser kitParser = factory.getKits();
+      SetMultimap<String, KeeperData> locations = parseShopKeepers(doc.getRootElement(), logger);
+
+      for (Element shop : XMLUtils.flattenElements(doc.getRootElement(), "shops")) {
+        Attribute shopName = XMLUtils.getRequiredAttribute(shop, "name");
+        List<Category> categories = Lists.newArrayList();
+
+        for (Element category : XMLUtils.getChildren(shop, "category")) {
+          Attribute categoryName = XMLUtils.getRequiredAttribute(category, "name");
+          Attribute categoryMaterial = XMLUtils.getRequiredAttribute(category, "material");
+          Material material = parseMaterial(categoryMaterial, category);
```

Comment:
```suggestion
          String categoryName = XMLUtils.getRequiredAttribute(category, "name").getValue();
          Material material = XMLUtils.parseMaterial(Node.fromRequiredAttr(category, "material"));
```

I'd consider just parsing category as an item, just like icons. That allows adding lore, or other extra stuff with little effort. If you need the "name" for identification purposes, use an id for the category (which you may auto-generate), not a name.